source/Mlos.Model.Services.UnitTests/UnitTests.cs

```diff
@@ -20,6 +22,211 @@
 
 namespace Mlos.Model.Services.UnitTests
 {
+    public class TestOptimizerServiceEncoderDecoder
+    {
+        private Client.OptimizerServiceEncoder encoder = new Client.OptimizerServiceEncoder();
+        private Client.OptimizerServiceDecoder decoder = new Client.OptimizerServiceDecoder();
+
+        // This function is required because Assert.Contains uses the == equivalence relation rather than .Equals()
+        private void AssertCollectionSubset(ReadOnlyCollection<object> list1, ReadOnlyCollection<object> list2)
+        {
+            foreach (var item1 in list1)

```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sequenceequal?view=net-5.0

Reply:
[zaporter/grpc 4a74b7b5] Refactored to use C# syntax tricks